index.d.ts

```diff
@@ -0,0 +1,59 @@
+import { ViewStyle, LayoutAnimationConfig } from "react-native";
+import { ReactNode } from "react";
+
+interface KeyboardAccessoryViewProps {
+  style?: ViewStyle;
+  animateOn?: "ios" | "android" | "all" | "none";
+  animationConfig?: (() => LayoutAnimationConfig) | LayoutAnimationConfig;
+  alwaysVisible?: boolean;
+  bumperHeight?: number;
+  visibleOpacity?: number;
+  onKeyboardShowDelay?: boolean | number;
+  hiddenOpacity?: number;
+  hideBorder?: boolean;
+  inSafeAreaView?: boolean;
+  androidAdjustResize?: boolean;
+  avoidKeyboard?: boolean;
+  children:
+    | (({ isKeyboardVisible }: { isKeyboardVisible: boolean }) => ReactNode)
+    | ReactNode;
+}
+
+export declare const KeyboardAccessoryView: React.ComponentType<KeyboardAccessoryViewProps>;
+
+type ArrowDirection = "down" | "up" | "right" | "left";
+
+interface KeyboardAccessoryNavigationProps extends KeyboardAccessoryViewProps {
+  doneButtonTitle?: string;
+  tintColor?: string;
+  doneButton?: ReactNode;
+  nextButton?: ReactNode;
+  previousButton?: ReactNode;
+  infoContainer?: ReactNode;
+  doneDisabled?: boolean;
+  nextDisabled?: boolean;
+  previousDisabled?: boolean;
+  doneHidden?: boolean;
+  nextHidden?: boolean;
+  previousHidden?: boolean;
+  accessoryStyle?: ViewStyle;
+  doneButtonStyle?: ViewStyle;
+  doneButtonTitleStyle?: ViewStyle;
+  infoMessageStyle?: ViewStyle;
+  previousButtonStyle?: ViewStyle;
+  nextButtonStyle?: ViewStyle;
+  nextButtonDirection?: ArrowDirection;
+  previousButtonDirection?: ArrowDirection;
+  onDone?: () => void;
+  onNext?: () => void;
+  onPrevious?: () => void;
+}
+
+export declare const KeyboardAccessoryNavigation: React.ComponentType<KeyboardAccessoryNavigationProps>;
+
+interface KeyboardAwareTabBarComponentProps {
+  TabBarComponent: React.ComponentType<any>;
+  [x: string]: any;
```

Comment:
Are you sure it's an any type?

Reply:
Do you have a suggestion for this one? It's not clear how to type this, given that the props are passed through: https://github.com/ardaogulcan/react-native-keyboard-accessory/blob/master/src/KeyboardAwareTabBarComponent.js#L45
It might be possible to do something with generics, but it's not so straight forward.

Reply:
Nope I don't, if it's not critical, we could left like this for now

Reply:
I think that's fair - this also seems to be one of the lesser used components in this package.